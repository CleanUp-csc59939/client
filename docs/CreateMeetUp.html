<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: CreateMeetUp</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: CreateMeetUp</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>CreateMeetUp<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="CreateMeetUp"><span class="type-signature"></span>new CreateMeetUp<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h4>
    

    



<div class="description">
    <p>returns page which contains a form to create an event</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_meetings_createMeetUp.jsx.html">components/meetings/createMeetUp.jsx</a>, <a href="components_meetings_createMeetUp.jsx.html#line13">line 13</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Return single event component</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="~onFinish"><span class="type-signature">(inner) </span>onFinish<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Calls the create event service with user inputs from the form after submission</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_meetings_createMeetUp.jsx.html">components/meetings/createMeetUp.jsx</a>, <a href="components_meetings_createMeetUp.jsx.html#line39">line 39</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="~updateSearch"><span class="type-signature">(async, inner) </span>updateSearch<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Updates the search index with the current event created</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_meetings_createMeetUp.jsx.html">components/meetings/createMeetUp.jsx</a>, <a href="components_meetings_createMeetUp.jsx.html#line24">line 24</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Banner.html">Banner</a></li><li><a href="CreateMeetUp.html">CreateMeetUp</a></li><li><a href="CustomSearch.html">CustomSearch</a></li><li><a href="Header.html">Header</a></li><li><a href="Home.html">Home</a></li><li><a href="Login.html">Login</a></li><li><a href="SingleEvent.html">SingleEvent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#authHeader">authHeader</a></li><li><a href="global.html#ConvertDate">ConvertDate</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.7</a> on Tue May 25 2021 00:59:11 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>