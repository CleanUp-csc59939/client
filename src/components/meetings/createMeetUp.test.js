import React from 'react';
import { shallow } from 'enzyme';
import CreateMeetUp from './createMeetUp';

describe('Create Event', () => {
  it('should render correctly in "debug" mode', async () => {
    const component = shallow(<CreateMeetUp debug />);
    expect(component).toMatchSnapshot();
  });
});
