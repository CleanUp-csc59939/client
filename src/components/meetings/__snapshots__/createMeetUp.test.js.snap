// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Event should render correctly in "debug" mode 1`] = `
<div>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Create MeetUp
  </h1>
  <div
    className="web"
  >
    <FormWeb
      onFinish={[Function]}
      onFinishFailed={[Function]}
    />
  </div>
</div>
`;
